assets/js/types/type-defs/payments.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { CartTotals } from './cart';
+import {
+	CartResponseBillingAddress,
+	CartResponseShippingAddress,
+} from './cart-response';
+import type { EmptyObjectType } from './objects';
+
+export interface Supports {
+	showSavedCards?: boolean;
+	showSaveOption?: boolean;
+	features?: string[];
+	// Deprecated, in favour of showSavedCards and showSaveOption
+	savePaymentInfo?: boolean;
+}
+
+// we assign a value in the class for supports.features
+export interface SupportsInConstructor extends Supports {
+	features: string[];
+}
+
+export interface CanMakePaymentArgument {
+	cartTotals: CartTotals;
+	cartNeedsShipping: boolean;
+	billingData: CartResponseBillingAddress;
+	shippingAddress: CartResponseShippingAddress;
+	selectedShippingMethods: Array< unknown >;
+	paymentRequirements: Array< string >;
+}
+
+export type CanMakePaymentReturnType =
+	| boolean
+	| Promise< boolean | { error: { message: string } } >;
```

Comment:
When is this ever a promise? I don't see anywhere in our code where we handle it, but maybe I'm missing something?

Reply:
Found it :)
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/e9fd11d53235cc6414daac1dcc84c67909bfb9c1/assets/js/base/context/providers/cart-checkout/payment-methods/use-payment-method-registration.ts#L113

Reply:
Yes, exactly, and it was also mentioned here https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/base/context/providers/cart-checkout/payment-methods/types.ts#L23-L25